python/cudf/cudf/core/series.py

```diff
@@ -1592,7 +1592,7 @@ def label_encoding(self, cats, dtype=None, na_sentinel=-1):
         codes = Series(cupy.arange(len(cats), dtype=dtype))
 
         value = DataFrame({"value": cats, "code": codes})
-        codes = DataFrame({"value": self.copy(), "order": order})
+        codes = DataFrame({"value": self.copy()._data.columns[0], "order": order})
```

Comment:
Do we need to protect against empty `_data` here?

Reply:
Just checked against an empty series, we seem to have `NumericalColumn` if it's empty/`None`:
```python
>>> s = cudf.Series([])
>>> s.factorize()
(Series([], dtype: int32), Series([], dtype: float64))
>>> s._data
ColumnAccessor(OrderedColumnDict([(None, <cudf.core.column.numerical.NumericalColumn object at 0x7fa98b19b910>)]), multiindex=False, level_names=(None,))
>>> s._data.columns[0]
<cudf.core.column.numerical.NumericalColumn object at 0x7fa98b19b910>
>>> s = cudf.Series(None)
>>> s
Series([], dtype: float64)
>>> s._data.columns[0]
<cudf.core.column.numerical.NumericalColumn object at 0x7fa98b19be10>
```

Would you want me to dodge it by having a check if the series is empty?

Reply:
Hmm, `DataFrame` and `Series` seem different in this aspect:

```
In [1]: import cudf

In [2]: a = cudf.DataFrame()

In [3]: a._data
Out[3]: ColumnAccessor(OrderedColumnDict(), multiindex=False, level_names=())

In [4]: a = cudf.Series()

In [5]: a._data
Out[5]: ColumnAccessor(OrderedColumnDict([(None, <cudf.core.column.numerical.NumericalColumn object at 0x7fa6b0eb67d0>)]), multiindex=False, level_names=(None,))
```

I'd rather we did the check on `._data` directly rather than at the Series level. In the future, it's possible we may change the implementation of an empty Series later to have an empty `_data`.

Reply:
I don't think it makes sense to have a `Series` without a `Column`, because a `Series` always has a dtype, whereas a DataFrame doesn't always have dtypes.